src/nwb2bids/_converters/_dataset_converter.py

```diff
@@ -264,6 +265,20 @@ def write_participants_metadata(self) -> None:
         # Deduplicate all rows of the frame
         deduplicated_data_frame = full_participants_data_frame.drop_duplicates(ignore_index=True)
```

Comment:
```suggestion
        deduplicated_data_frame = full_participants_data_frame.drop_duplicates(ignore_index=True).copy()
```

This will remove the `SettingWithCopyWarning`.